lnrpc/routerrpc/router_backend.go

```diff
@@ -354,15 +363,22 @@ func (r *RouterBackend) QueryRoutes(ctx context.Context,
 
 // getSuccessProbability returns the success probability for the given route
 // based on the current state of mission control.
-func (r *RouterBackend) getSuccessProbability(rt *route.Route) float64 {
+func (r *RouterBackend) getSuccessProbability(
+	rt *route.Route) (float64, error) {
+
 	fromNode := rt.SourcePubKey
 	amtToFwd := rt.TotalAmount
 	successProb := 1.0
 	for _, hop := range rt.Hops {
 		toNode := hop.PubKeyBytes
 
+		capacity, err := r.FetchPairCapacity(fromNode, toNode)
```

Comment:
Such a dramatic slowdown for an in-memory operation. Did you pull a cpu profile to see what's happening?

Reply:
Interesting. So probably fixable by making a specialized capacity lookup on `GraphCache`? Don't know if it is worth it. It is better separation of concerns though.

Reply:
Yes, I'll try to see if we can have something performant there.

Reply:
we decided to go with the uniform policies approach as it avoids a second traversal of a node's peers